esphome/components/ethernet/ethernet_component.h

```diff
@@ -83,6 +83,7 @@ class EthernetComponent : public Component {
 #endif
   void set_type(EthernetType type);
   void set_manual_ip(const ManualIP &manual_ip);
+  void set_fixed_mac(const std::vector<uint8_t> &mac) { fixed_mac_ = mac; }
```

Comment:
A fixed size Array would be better than a vector here since we know it will always have a length of 6.
```suggestion
  void set_fixed_mac(const std::vector<uint8_t> &mac) { this->fixed_mac_ = mac; }
```

Reply:
@sphen13 you fixed the this-> but didnt address "A fixed size Array would be better than a vector here since we know it will always have a length of 6."

Reply:
@swoboda1337 sorry! was too fast here. i hope my changes now satisfy both suggestions.